discovery_engine_core/core/src/storage.rs

```diff
@@ -65,7 +70,11 @@ pub(crate) trait FeedScope {
     async fn fetch(&self) -> Result<Vec<ApiDocumentView>, Error>;
 
     // helper function. will be replaced later by move_from_stacks_to_feed
-    async fn store_documents(&self, documents: &[NewDocument]) -> Result<(), Error>;
+    async fn store_documents(
+        &self,
+        documents: &[NewDocument],
+        stack_ids: &HashMap<document::Id, stack::Id>,
```

Comment:
❓ can't we instead add a `stack_id` field to `NewDocument`? this seems less work than to collect the doc-to-stack-id mapping.

Reply:
I wanted to avoid this as `NewDocument` is currently not meant to be specific to the sub-system which added it to the document (i.e. sub-system  == feed, search, etc. we don't have terminology for this yet).

And the stack id and how it's handled is quite specific to the feed.

Furthermore when we have the full database soon in sqlite we will add new documents to a specific stack, i.e. all their stack ids need to be the same.  For which I would like to use `.stack().store_new_documents(StackId, &[NewDocument])` as this most reliable represents the mechanism involved.

And for the current situation using the hash map is the simplest solution which touches the least amount of unrelated code.